app/handlers/handlers_groups.py

```diff
@@ -0,0 +1,85 @@
+from datetime import timedelta
+
+import kopf
+
+from app.api import TwingateAPIClient
+from app.api.exceptions import GraphQLMutationError
+from app.crds import GroupStatusUserID, TwingateGroupCRD
+from app.handlers import fail, success
+
+
+@kopf.on.resume("twingategroup")
+@kopf.on.create("twingategroup")
+@kopf.on.update("twingategroup")
+def twingate_group_create_update(body, spec, logger, memo, patch, **kwargs):
+    logger.info("twingate_group_reconciler: %s", spec)
+    settings = memo.twingate_settings
+    client = TwingateAPIClient(settings)
+
+    if diff := kwargs.get("diff"):
+        logger.info("Diff: %s", diff)
+        # If ID changed from `None to value we just created it no need to update
+        # diff is `(('add', ('spec', 'id'), None, 'R3JvdXA6MjAxNjc4OQ=='),)`
+        if (
+            len(diff) == 1
+            and diff[0][0] == "add"
+            and diff[0][1] == ("spec", "id")
+            and diff[0][2] is None
```

Comment:
nit: maybe `diff[0][:3] == ("add", ("spec", "id"), None)` like in the [resource update handler](https://github.com/Twingate/kubernetes-operator/blob/d143995d46f1a381d0571dfb05df9843d919abcd/app/handlers/handlers_resource.py#L51)